client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,45 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * If a Stream's Retention Policy {@link RetentionPolicy} is set to 'CONSUMPTION' based,
+     * the Reader Group needs to register with the Stream as a Subscriber
+     * and periodically publish retention StreamCuts either manually or at checkpoints.
+     * This can be done by setting retentionConfig to 'CONSUMPTION_BASED_USER_STREAMCUT' or
+     * 'CONSUMPTION_BASED_AT_LAST_CHECKPOINT'.
+     * If the Stream's RetentionPolicy is not Consumption based, retentionConfig should be 'NO_IMPACT'.
```

Comment:
>  * If the Stream's RetentionPolicy is not Consumption based, retentionConfig should be 'NO_IMPACT'.

"Consumption based" is not clear. Please link to the exact setting explicitly. 
"should be 'NO_IMPACT'" is a meaningless detective. We should say that this setting will have no effect regardless of what it is set to.

Reply:
Consumption based has just been explained 2 lines earlier along with the link.
Nevertheless added the link again

Reply:
> Consumption enum value will be incorrect.

Will it throw an exception? If so document that. If not it's not "invalid", in which case what it does should be described.

Reply:
> I was assuming that doing so would be fine and would work as expected on the stream with it enabled and have no effect on the other stream. 

Yes. Thats right. 
It is not possible to provide the explanation for all these varied combinations of Stream and Reader Group options and their impact in the javadoc and still keep it succinct. 
I will add these details in the User Documentation.